core/src/test/java/com/graphhopper/routing/subnetwork/PrepareRoutingSubnetworksTest.java

```diff
@@ -226,35 +226,38 @@ GraphHopperStorage createSubnetworkTestStorageWithOneWays(String flagEncodersStr
 
     @Test
     public void testRemoveSubNetworks_withOneWays() {
-        GraphHopperStorage g = createSubnetworkTestStorageWithOneWays("car");
-        FlagEncoder encoder = g.getEncodingManager().fetchEdgeEncoders().iterator().next();
+        EncodingManager em = createEncodingManager("car");
+        FlagEncoder encoder = em.fetchEdgeEncoders().iterator().next();
+        GraphHopperStorage g = createSubnetworkTestStorageWithOneWays(em, encoder);
         assertEquals(11, g.getNodes());
 
         PrepareRoutingSubnetworks.PrepareJob job = createJob(g.getEncodingManager(), encoder, NO_TURN_COST_PROVIDER);
         PrepareRoutingSubnetworks instance = new PrepareRoutingSubnetworks(g, Collections.singletonList(job)).
-                setMinNetworkSize(3);
-        int removed = instance.setSubnetworks(job);
+                setMinNetworkSize(2);
+        int subnetworkEdges = instance.doWork();
 
-        // the (7) and the (5,6) components get removed -> 2 remaining components and 3 removed edges plus the 2 connecting oneway edges
-        // note that the subnetworkEV per profile is one bit per *edge*. Before we used the encoder$access with 2 bits and got more fine grained response here (8 removed *edgeKeys*)
-        assertEquals(5, removed);
+        // the (7) and the (5,6) components become subnetworks -> 2 remaining components and 3 subnetwork edges
+        // note that the subnetworkEV per profile is one bit per *edge*. Before we used the encoder$access with 2 bits
+        // and got more fine grained response here (8 removed *edgeKeys*)
+        assertEquals(3, subnetworkEdges);
+        assertEquals(IntArrayList.from(4, 5, 6), getSubnetworkEdges(g, encoder));
     }
 
     @Test
-    public void testAddEdgesAfterwards() {
-        GraphHopperStorage g = createSubnetworkTestStorageWithOneWays("car");
-        FlagEncoder encoder = g.getEncodingManager().fetchEdgeEncoders().iterator().next();
+    public void testRemoveSubNetworks_withOneWays_higherMinNetworkSize() {
+        EncodingManager em = createEncodingManager("car");
+        FlagEncoder encoder = em.fetchEdgeEncoders().iterator().next();
+        GraphHopperStorage g = createSubnetworkTestStorageWithOneWays(em, encoder);
         assertEquals(11, g.getNodes());
 
         PrepareRoutingSubnetworks.PrepareJob job = createJob(g.getEncodingManager(), encoder, NO_TURN_COST_PROVIDER);
         PrepareRoutingSubnetworks instance = new PrepareRoutingSubnetworks(g, Collections.singletonList(job)).
                 setMinNetworkSize(3);
-        int removed = instance.setSubnetworks(job);
+        int subnetworkEdges = instance.doWork();
 
-        assertEquals(5, removed);
-        assertTrue(isConsistent(g));
-        g.edge(7, 8);
-        assertTrue(isConsistent(g));
+        // this time also the (8,9,10) component is a subnetwork
```

Comment:
I think since we are not removing edges or nodes anymore there is no need to check the 'consistency' of the graph or whether or not we can add edges after subnetwork search (which we do not do in production anyway). All we need to test is that the right edges get marked as subnetwork edges.